### PR TITLE
Use TaskCache class from Microsoft.Extensions.TaskCache.Sources

### PR DESCRIPTION
**Pull request by pakrym**
Instead of Task.FromResult(0)

@pranavkm

### CODE REVIEW

File: test/Microsoft.AspNetCore.Routing.Tests/RoutingBuilderExtensionsTest.cs [hunk 1]
```diff
@@ -6,8 +6,8 @@
 using Microsoft.AspNetCore.Builder.Internal;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Routing;
-using Microsoft.AspNetCore.Routing.Internal;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Routing.Tests/RoutingBuilderExtensionsTest.cs]: Could we use `Task.FromResult` in tests? It avoids type conflict issues if we ever add an `InternalsVisibleTo` in the source project.

**Comment by pranavkm** [comment]
:watch: for the question.

**Comment by pakrym** [comment]
Switched tests to use `Task.FromResult()`

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**